src/ZODB/DB.py

```diff
@@ -438,12 +444,10 @@ def __init__(self, storage,
                 DeprecationWarning, 2)
             storage.tpc_vote = lambda *args: None
 
-        if IMVCCStorage.providedBy(storage):
-            temp_storage = storage.new_instance()
-        else:
-            temp_storage = storage
+        temp_storage = self._mvcc_storage.new_instance()
```

Comment:
Somewhere you recently said something about how the way IMVCCStorage clones itself is weird. I agree :) an IMVCCStorageFactory could clean this up and make the distinctions and responsibilities clearer.